IBAnalyzer/Analyzers/ConnectionAnalyzer.swift

```diff
@@ -7,25 +7,127 @@
 //
 
 import Foundation
+import SourceKittenFramework
+
+struct Violation {
+    var name: String
+    var line: Int
+    var column: Int
+    var url: URL?
+    var isOptional: Bool
+
+    init(name: String, line: Int, column: Int, url: URL? = nil, isOptional: Bool = false) {
+        self.name = name
+        self.line = line
+        self.column = column
+        self.url = url
+        self.isOptional = isOptional
+    }
+
+    init(name: String, file: File, offset: Int64, isOptional: Bool = false) {
+        let fileOffset = Violation.getLineColumnNumber(of: file, offset: Int(offset))
+        var url: URL?
+        if let path = file.path {
+            url = URL(string: path)
```

Comment:
Better to use `URL(fileURLWithPath:)` as `URL(string:)` sometimes returns `nil` for paths.